src/server/dfly_main.cc

```diff
@@ -161,9 +163,28 @@ struct VersionMonitor {
   }
 
  private:
+  bool IsVersionOutdated(std::string_view remote, std::string_view current) const;
   void RunTask(SSL_CTX* ssl_ctx);
 };
 
+bool VersionMonitor::IsVersionOutdated(const std::string_view remote,
+                                       const std::string_view current) const {
+  const std::vector<absl::string_view> remote_xyz = absl::StrSplit(remote, ".");
+  const std::vector<absl::string_view> current_xyz = absl::StrSplit(current, ".");
+  CHECK_EQ(remote_xyz.size(), current_xyz.size());
```

Comment:
I wouldn't crash if the remote responds with a weird version. A bad push on our side can effectively crash all Dragonfly instances in the world :worried: 
I would perhaps `LOG` some warning and return that the version is not outdated.